### PR TITLE
feat: Action to compute a sha256 of a file

### PR DESCRIPTION
**Pull request by laurentsimon**
This is part of https://github.com/slsa-framework/slsa-github-generator/issues/26

This PR defines an action to compute the sha256 of a file.

I will send follow-up PRs to use these it as part of artifact verification and for other uses in the re-usable workflows. We may later convert this code to TS or another programming language.

### CODE REVIEW

File: .github/actions/compute-sha256/action.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+runs:
+  using: "composite"
+  steps:
+    - name: Compute the sha256
+      id: compute
+      env:
+        UNTRUSTED_PATH: "${{ inputs.path }}"
+      run: |
+
+        set -euo pipefail
```
Comment (ianlewis) [inline .github/actions/compute-sha256/action.yml:21]: I don't think we need this if we're just going to call out to another script. We can just do

```yaml
run: ./.github/actions/compute-sha256.sh
```

**Comment by ianlewis** [comment]
How much value do we get out of just computing the sha256? I thought maybe our action would combine the generate sha and upload steps. And another action could combine the download and verify steps. I'm curious what you had in mind?

File: .github/actions/compute-sha256/action.yml (hunk 1 above)
Comment (laurentsimon) [reply .github/actions/compute-sha256/action.yml:21]: yeah Ive been thinking about this too. I'm not yet 100% sure whether this action will be replaced/removed in the end. I was thinking that may be we will replace the script code with TS. 
But maybe it's better to use the script in this PR instead. Shall I do that?

Comment (laurentsimon) [reply .github/actions/compute-sha256/action.yml:21]: Let's go with the script option then.

Comment (laurentsimon) [reply .github/actions/compute-sha256/action.yml:21]: Well, let me know how you feel before I proceed :-)

Comment (ianlewis) [reply .github/actions/compute-sha256/action.yml:21]: I was just referring to the `set -euo pipefail` part.

**Review by ianlewis — APPROVED** [review]

File: .github/actions/compute-sha256/action.yml (hunk 1 above)
Comment (laurentsimon) [reply .github/actions/compute-sha256/action.yml:21]: sorry, I mis-understood. Gotcha.

Comment (laurentsimon) [reply .github/actions/compute-sha256/action.yml:21]: done, thanks

**Pull request merged**